Orange/widgets/utils/tests/concurrent_example.py

```diff
@@ -0,0 +1,147 @@
+# pylint: disable=too-many-ancestors
+from typing import Optional
+from types import SimpleNamespace as namespace
+from functools import partial
+
+import numpy as np
+
+from Orange.data import Table
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.utils.concurrent import TaskState, ConcurrentWidgetMixin
+from Orange.widgets.utils.widgetpreview import WidgetPreview
+from Orange.widgets.visualize.utils.widget import OWDataProjectionWidget
+
+
+class Results(namespace):
+    embedding = None  # type: Optional[np.ndarray]
+
+
+class Runner:
```

Comment:
`run` can be its own method, there is no need for it to be wrapped in a class.

The reason I made a runner class for t-SNE was that I had several methods and I wanted to group them together. Here, you have one method so a function is cleaner.